### PR TITLE
Feature/use user profile data object

### PR DESCRIPTION
**Pull request by n13**
Support for all networks:

Always process transactions and http requests on the network the user is on. 

Never default to Telos. Except when reading in old profiles, we assume they're on Telos. 

Users who have testnet accounts just need to log out and log back in.

**I have tested all changes** 

Yes.

### CODE REVIEW

File: lib/core/crypto/dart_esr/src/signing_request_manager.dart [hunk 1]
```diff
@@ -7,7 +7,7 @@ import 'package:hypha_wallet/core/crypto/dart_esr/src/models/request_signature.d
 import 'package:hypha_wallet/core/crypto/dart_esr/src/utils/base64u.dart';
```
Comment (n13) [inline lib/core/crypto/dart_esr/src/signing_request_manager.dart:7]: A lot of these changes - I moved Network enum into its own file

**Comment by gguijarro-c-chwy** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>:exclamation: PR has more than 500 line changes </td>
    </tr>
  </tbody>
</table>



> Pull Request size seems relatively large. If Pull Request contains multiple changes, split each into separate PR will helps faster, easier review.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 266cb61828d584ef1b8cd33a0e281cd826816319
</p>

File: lib/core/di/services_module.dart [hunk 2]
```diff
@@ -7,7 +7,9 @@ Future<void> _registerServicesModule() async {
```
Comment (n13) [inline lib/core/di/services_module.dart:7]: I think we need to remove networking manager as a singleton since we have 1 for each network

**Comment by n13** [comment]
Tested all functions - shipping.

File: lib/ui/onboarding/edit_account/interactor/edit_account_event.dart [hunk 3]
```diff
@@ -5,5 +5,5 @@ class EditAccountEvent with _$EditAccountEvent {
   const factory EditAccountEvent.initial() = _Initial;
   const factory EditAccountEvent.onNextPressed(InviteLinkData inviteLinkData) = _OnNextPressed;
   const factory EditAccountEvent.clearPageCommand() = _ClearPageCommand;
-  const factory EditAccountEvent.onAccountChange(String value) = _OnAccountChange;
+  const factory EditAccountEvent.onAccountChange(String value, Network network) = _OnAccountChange;
```
Comment (gguijarro-c-chwy) [inline lib/ui/onboarding/edit_account/interactor/edit_account_event.dart:8]: @n13  this is not needed. You can get the network from the state inside the bloc, no need to pass it from the UI

**Pull request merged**